frontend/public/components/secrets/create-secret/PullSecretCredentialsForm.tsx

```diff
@@ -0,0 +1,111 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button } from '@patternfly/react-core';
+import { MinusCircleIcon } from '@patternfly/react-icons/dist/esm/icons/minus-circle-icon';
+import { PlusCircleIcon } from '@patternfly/react-icons/dist/esm/icons/plus-circle-icon';
+import { AUTHS_KEY, PullSecretCredentialEntry, PullSecretData } from '.';
+import { usePullCredentialsFormSecretEntries } from './usePullCredentialsFormSecretEntries';
+
+const newImageSecretEntry = (): PullSecretCredential => ({
+  address: '',
+  username: '',
+  password: '',
+  email: '',
+  auth: '',
+  uid: _.uniqueId(),
+});
+
+export const PullSecretCredentialsForm: React.FC<PullSecretCredentialsFormProps> = ({
+  onChange,
+  pullSecretData,
+}) => {
+  const { t } = useTranslation();
+  const [secretEntriesArray, setSecretEntriesArray] = usePullCredentialsFormSecretEntries(
+    pullSecretData?.[AUTHS_KEY] || pullSecretData,
+    onChange,
+  );
+
+  const onEntriesChanged = (secretEntries: PullSecretCredential[]) => {
+    setSecretEntriesArray(secretEntries);
+  };
+
+  const updateEntry = (updatedEntry, entryIndex: number) => {
+    const updatedSecretEntriesArray = secretEntriesArray.map((entry, index) => {
+      if (index === entryIndex) {
+        return {
+          uid: entry.uid,
+          ...updatedEntry,
+        };
+      }
+      return entry;
+    });
```

Comment:
nit, this can be simplified to a ternary expression and an arrow function shortcut
```suggestion
    const updatedEntries = entries.map(
      (entry, index) => index === entryIndex
        ? { uid: entry.uid, ...updatedEntry } 
        : entry,
    );
```